### ISSUE TITLE
company produces problems with " in eshell

### ISSUE THREAD

**Issue by Poquaruse** [company-mode/company-mode#409]
Hi all,

if company is enabled and an eshell is opened, the following error is logged to _Messages_ if I type just " or a command like hg commit -m " in the eshell. 

"Completion function pcomplete-completions-at-point uses a deprecated calling convention"

Furthermore, and very annoyingly, a tab is inserted... Disabling company mode makes this problem go away.

Is this a known problem? Any ideas on how to fix it?

Best regards

**Comment by dgutov** [comment]
> uses a deprecated calling convention

Well, it does. That's a known "FIXME" at the beginning of the definition of `pcomplete-completions-at-point` (the second one). Not sure where the exact issue lies (in pcomplete or Eshell), but I'm sure the Emacs core would enjoy a patch.

> Any ideas on how to fix it?

As a workaround, you can remove `pcomplete-completions-at-point` from `completion-at-point-functions` in `eshell-mode-hook`.

**Comment by Poquaruse** [comment]
Thank you for the explanations! So it does indeed make sense that this message appears. ;-)

Unfortunately, I don't know any Lisp beyond the usual stuff you need in init.el. Therefore, I won't be able to fix anything. :-( 

I might however try to remove the said function (?!) from the list (?!) when the hook is called (?!) in my init.el...

Thanks again and best regards

**Comment by dgutov** [comment]
> ?! ?! ?!

Ok, here's the code. :)

```el
(defun my-eshell-remove-pcomplete ()
  (remove-hook 'completion-at-point-functions #'pcomplete-completions-at-point t))

(add-hook 'eshell-mode-hook #'my-eshell-remove-pcomplete)
```

You should really learn Lisp, though. Emacs is considerably less cool when you can't program it.

**Comment by Poquaruse** [comment]
Thanks! I even managed to make a lambda function out of it. :) 

In my favorites there's an elisp tutorial. I've been wanting to start programming it for quite some time now. Guess, there's no way around it now.

Best regards and thanks for helping me out!

**Comment by dgutov** [comment]
> I even managed to make a lambda function out of it.

Note that using lambda functions with hooks is not particularly recommended: it's harder to read the hook's value, and it's harder to remove the function from the hook. But of course it's up to you.

> Best regards and thanks for helping me out!

No problem.

**Comment by chameco** [comment]
Since this is the only place I've seen this issue discussed, I figured I'd comment here. The issue is in `eshell-complete-parse-arguments`, which is used as the `pcomplete-parse-arguments-function`. `eshell-complete-parse-arguments` inserts a tab character whenever no completion is found, presumably to allow the user to press TAB for both completion and inserting tabs - what a hack!

There isn't even any way to configure this, so I worked around it by just redefining `eshell-complete-parse-arguments` with all of the `(insert-and-inherit "\t")` expressions removed.

Once I did this, my Company + Eshell experience was *dramatically* improved.
